src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -462,17 +461,9 @@ private async ValueTask ProcessDataWrites()
                 // Stream state will move to RequestProcessingStatus.ResponseCompleted
                 _responseCompleteTaskSource.SetResult(flushResult);
 
-                if (readResult.IsCompleted)
-                {
-                    // Successfully read all data. Wait here for the stream to be reset.
-                    await new ValueTask(_resetAwaitable, _resetAwaitable.Version);
-                    _resetAwaitable.Reset();
-                }
-                else
-                {
-                    // Stream was aborted.
-                    break;
-                }
+                // Wait here for the stream to be reset or disposed.
+                await new ValueTask(_resetAwaitable, _resetAwaitable.Version);
+                _resetAwaitable.Reset();
```

Comment:
Does the fact this is pushed mean that this version of the change also works reliably?

Reply:
I ran it for 45 minutes without any failures.